pkg/server/controllers.go

```diff
@@ -345,15 +428,14 @@ func (s *Server) installTenancyLogicalClusterController(ctx context.Context, con
 		return err
 	}
 
-	controller := tenancylogicalcluster.NewController(
-		kubeClusterClient,
-		s.KcpSharedInformerFactory.Core().V1alpha1().LogicalClusters(),
-		s.KubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings(),
-	)
-
 	return s.registerController(&controllerWrapper{
 		Name: tenancylogicalcluster.ControllerName,
 		Runner: func(ctx context.Context) {
+			controller := tenancylogicalcluster.NewController(
+				kubeClusterClient,
+				s.KcpSharedInformerFactory.Core().V1alpha1().LogicalClusters(),
+				s.KubeSharedInformerFactory.Rbac().V1().ClusterRoleBindings(),
+			)
```

Comment:
🔴  What is the semantics here now in the Runner? Do we ensure the informer factor is started after all the runners? Looking at the code this can't be the case: `controller.Start` blocks. This means we have a race of the individual informers not being started if the factory is faster than the runners.

Reply:
In few controllers the wait for informer cache to sync logic is implemented using wait method before start.Can we just add this for all the controllers for which informers to be sycned before start ?

https://github.com/sankar17/kcp/blob/sankar17/rest-cfg-timeout-and-workspace-ctrl-leader-election-fix/pkg/server/controllers.go#L719-L732

https://github.com/sankar17/kcp/blob/sankar17/rest-cfg-timeout-and-workspace-ctrl-leader-election-fix/pkg/server/controllers.go#L912-L918

https://github.com/sankar17/kcp/blob/sankar17/rest-cfg-timeout-and-workspace-ctrl-leader-election-fix/pkg/server/controllers.go#L991-L997

https://github.com/sankar17/kcp/blob/sankar17/rest-cfg-timeout-and-workspace-ctrl-leader-election-fix/pkg/server/controllers.go#L1217-L1225

https://github.com/sankar17/kcp/blob/sankar17/rest-cfg-timeout-and-workspace-ctrl-leader-election-fix/pkg/server/controllers.go#L1254-L1261

https://github.com/sankar17/kcp/blob/sankar17/rest-cfg-timeout-and-workspace-ctrl-leader-election-fix/pkg/server/controllers.go#L1467-L1478


CC: @yastij @palnabarun